### PR TITLE
Delete unused costing code

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Resolves #SL-1068

### Description
Delete costing code that's no longer used. Discussion on this can be found [here](https://github.com/dbt-labs/metricflow/pull/801).

### CODE REVIEW

File: metricflow/model/semantics/metric_lookup.py [hunk 1]
```diff
@@ -127,6 +127,8 @@ def measures_for_metric(
                     column_association_resolver=column_association_resolver,
                 ).create_from_where_filter_intersection(input_measure.filter),
                 alias=input_measure.alias,
+                join_to_timespine=input_measure.join_to_timespine,
+                fill_nulls_with=input_measure.fill_nulls_with,
```
Comment (tlento) [inline metricflow/model/semantics/metric_lookup.py:131]: These are new..... I'm guessing from another branch? Let's revert these from this PR before merge.

File: metricflow/plan_conversion/dataflow_to_sql.py [hunk 2]
```diff
@@ -673,6 +673,7 @@ def visit_compute_metrics_node(self, node: ComputeMetricsNode) -> SqlDataSet:
                 else:
                     expr = metric.name
                 # Use a column reference to improve query optimization.
+                # Update metric expr to use coalesce; when to join to time spine??
```
Comment (tlento) [inline metricflow/plan_conversion/dataflow_to_sql.py:676]: Revert, probably?

File: metricflow/specs/specs.py [hunk 3]
```diff
@@ -463,6 +463,8 @@ class MetricInputMeasureSpec(SerializableDataclass):
     measure_spec: MeasureSpec
     constraint: Optional[WhereFilterSpec] = None
     alias: Optional[str] = None
+    join_to_timespine: bool = False
+    fill_nulls_with: Optional[int] = None
```
Comment (tlento) [inline metricflow/specs/specs.py:467]: And this

File: metricflow/test/snapshots/test_data_warehouse_tasks.py/data_warehouse_validation_manifest/BigQuery/test_build_metric_tasks__query0.sql [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
 -- Compute Metrics via Expressions
 SELECT
-  metric_time__day
-  , SUM(count_dogs) AS count_dogs
-FROM (
-  -- Read Elements From Semantic Model 'animals'
-  -- Metric Time Dimension 'ds'
-  -- Pass Only Elements:
-  --   ['count_dogs', 'metric_time__day']
+  date_day,
```
Comment (tlento) [inline metricflow/test/snapshots/test_data_warehouse_tasks.py/data_warehouse_validation_manifest/BigQuery/test_build_metric_tasks__query0.sql:4]: This doesn't belong here either, right?

**Review by tlento — COMMENTED** [review]
Is that BigQuery change from another branch or here?

File: metricflow/model/semantics/metric_lookup.py (hunk 1 above)
Comment (courtneyholcomb) [reply metricflow/model/semantics/metric_lookup.py:131]: dammit I didn't look at my PR after putting it up 😅  redo time

**Pull request closed without merging**